src/modules/hooks/useBlockExplorer.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback } from 'react';
+import { useChains } from 'wagmi';
+
+type ChainEntityType = 'address' | 'tx' | 'token';
+
+interface IChainEntity {
+    type: ChainEntityType;
+    chainId?: number;
+    id: string;
+}
+
+export const useBlockExplorer = () => {
+    const chains = useChains();
+
+    const getChainEntityUrl = useCallback(
```

Comment:
What if we pass the `type`, `chainId` and `id` parameters directly to the `useBlockExplorer` hook and return the URL instead of returning a function?